### ISSUE TITLE
Test Fails - LimitExceededException in TestRunner

### ISSUE THREAD

**Issue by mcawilcox** [aws-solutions/distributed-load-testing-on-aws#105]
**Describe the bug**
When starting a test, it (almost immediately fails), resulting in a FAILED status

**To Reproduce**
Start a test. It can be a newly-created test, or a previously-run test.
We're mostly running tests that should be a single task, and 100-200 users, with an uploaded Jmeter zip file.

Until earlier today, these tests would run as expected. In the process of amending the scripts to test a new environment, we had to fix a few faults, so cancelled a sequence of 5 tests over 40 minutes.

The next test failed to start, as have all attempts to execute since.
 
**Expected behavior**
Until earlier today, these tests would run as expected, executing, and returning results.

**Please complete the following information about the solution:**
- [x] Version:  **v2.0.2**
- [x] Region: **eu-west-2**
- [x] Was the solution modified from the version published on this repository? **No**
- [ ] If the answer to the previous question was yes, are the changes available on GitHub? **n/a**
- [ ] Have you checked your [service quotas](https://docs.aws.amazon.com/general/latest/gr/aws_service_limits.html) for the services this solution uses? **No**
- [x] Were there any errors in the CloudWatch Logs? **Yes**

**Screenshots**

**Additional context**
Looking through the lambda's, we could see that the Task Runner was showing errors.
The logfile for each failure shows a couple of lines:

```
2022-10-21T18:02:41.798Z	555c8a72-3c50-4c58-af6c-94ecbebd0fd3	INFO	
{
    "scenario": {
        "testId": "kWU8h4nflx",
        "taskCount": 1,
        "testType": "jmeter",
        "fileType": "zip"
    },
    "isRunning": false,
    "numTasksRunning": 1
}
```
```
2022-10-21T18:02:42.439Z	555c8a72-3c50-4c58-af6c-94ecbebd0fd3	ERROR	LimitExceededException: Resource limit exceeded.
    at Request.extractError (/var/runtime/node_modules/aws-sdk/lib/protocol/json.js:52:27)
    at Request.callListeners (/var/runtime/node_modules/aws-sdk/lib/sequential_executor.js:106:20)
    at Request.emit (/var/runtime/node_modules/aws-sdk/lib/sequential_executor.js:78:10)
    at Request.emit (/var/runtime/node_modules/aws-sdk/lib/request.js:686:14)
    at Request.transition (/var/runtime/node_modules/aws-sdk/lib/request.js:22:10)
    at AcceptorStateMachine.runTo (/var/runtime/node_modules/aws-sdk/lib/state_machine.js:14:12)
    at /var/runtime/node_modules/aws-sdk/lib/state_machine.js:26:10
    at Request.<anonymous> (/var/runtime/node_modules/aws-sdk/lib/request.js:38:9)
    at Request.<anonymous> (/var/runtime/node_modules/aws-sdk/lib/request.js:688:12)
    at Request.callListeners (/var/runtime/node_modules/aws-sdk/lib/sequential_executor.js:116:18) {
  code: 'LimitExceededException',
  time: 2022-10-21T18:02:42.379Z,
  requestId: 'ab648925-57de-4fb5-814c-522604bb6a7c',
  statusCode: 400,
  retryable: false,
  retryDelay: 28.537775678741983
}
```
The stack trace doesn't help (me) identify what limit has been exceeded. Some googling suggests it might be DynamoDB.

Looking in the DynamoDB table, there's nothing special about the last of the 5 cancelled tasks, compared to the previous 4.
The newly-failed tasks all get updated with a status of failed.
There's no sign of anything amiss in the DynamoDB table usage metrics.

Looking in ECS, I can see no tasks running in the cluster (nor any stopped tasks). In any event, the FAILED result turns up faster than the time it takes for an ECS task to move through provisioning and pending states.

Any ideas where to go looking?

**Comment by mcawilcox** [comment]
I did a little bit of debugging of the TaskRunner lambda, to see where it bombs out.
The issue is the line
`
await cloudwatchLogs.putMetricFilter(metricFilterParams).promise();
`

We had 25 old tests, which meant we had reached the limit of 100 metric filters on a log group.
It seems that limit cannot be increased.

I'll leave this issue open, even though I now know the cause, as I suspect there's an improvement that can be made here ... at least in presenting this error.

**Comment by G-Lenz** [comment]
Hi @mcawilcox 

This issue was addressed in the newest version (v3.0.0).

We now delete the metric filter each time a test is completed as you can see in the following section of code:

https://github.com/aws-solutions/distributed-load-testing-on-aws/blob/f616d634b352891c386d284a0f02f21e8404c5cd/source/results-parser/lib/parser/index.js#L433
